internal/webhooks/vspheremachinetemplate.go

```diff
@@ -84,6 +84,13 @@ func (webhook *VSphereMachineTemplateWebhook) ValidateCreate(_ context.Context,
 			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "guestSoftPowerOffTimeout"), spec.GuestSoftPowerOffTimeout, "should be greater than 0"))
 		}
 	}
+	for _, device := range spec.PciDevices {
+		hasVGPU := device.VGPUProfile != ""
+		hasPCI := device.DeviceID != nil && device.VendorID != nil
+		if (hasPCI && hasVGPU) || (!hasPCI && !hasVGPU) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices"), spec.PciDevices, "should have either deviceID + vendorID or vgpuProfile"))
```

Comment:
```suggestion
	for i, device := range spec.PciDevices {
		hasVGPU := device.VGPUProfile != ""
		hasPCI := device.DeviceID != nil && device.VendorID != nil
		if (hasPCI && hasVGPU) || (!hasPCI && !hasVGPU) {
			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d, i)), device, "should have either deviceID + vendorID or vgpuProfile"))
```

Reply:
Rewrite it to something like this? Which has better optics in your opinion?
```go
if device.VGPUProfile == "" {
  if device.DeviceID == nil || device.VendorID == nil {
    allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d", i)), device, "should have both deviceId and vendorId set"))
  
  }
} else {
  if device.DeviceID != nil || device.VendorID != nil {
    allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d", i)), device, "should have either deviceId + vendorId or vgpuProfile"))
  
  }
}
```

```go
if device.VGPUProfile == "" && (device.DeviceID == nil || device.VendorID == nil) {
  allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d", i)), device, "should have both deviceId and vendorId set"))
} else if device.VGPUProfile != "" && (device.DeviceID != nil || device.VendorID != nil) {
  allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d", i)), device, "should have either deviceId + vendorId or vgpuProfile"))
}
```